File: RxRetrofitApp/japp/src/main/java/com/woowrale/rxretrofit/japp/ui/base/BasePresenter.java
package com.woowrale.rxretrofit.japp.ui.base;

import android.support.annotation.NonNull;

public abstract class BasePresenter<T extends BasePresenter.View> {

    private T mView;
    private static final String TAG = BasePresenter.class.getName();

    protected T getView() {
        return mView;
    }

    public void setView(@NonNull T view) {
        mView = view;
    }

    public interface View {

        void showMessageError();

        void showMessageResult();
    }
}

File: RxKotlinRetrofit/app/src/main/java/com/woowrale/rxkotlinretrofit/ui/base/BasePresenter.kt
package com.woowrale.rxkotlinretrofit.ui.base

abstract class BasePresenter<T : BasePresenter.View> {

    private var mView: T? = null

    protected fun getView(): T? {
        return mView
    }

    fun setView(view: T?) {
        mView = view
    }

    interface View {

        fun showMessageError()

        fun showMessageResult()
    }
}